lib/mspec/runner/actions/constants_leak_checker.rb

```diff
@@ -0,0 +1,86 @@
+class ConstantsLeakChecker
+  attr_accessor :constants_new
+
+  def initialize
+    @constants_before = Module.constants
+  end
+
+  def check
+    @constants_after = Module.constants
+    @constants_new = @constants_after - @constants_before
+    @constants_new.empty?
+  end
+end
+
+class ConstantsLockFile
+  LOCK_FILE_NAME = '.mspec.constants.lock'
```

Comment:
Where would you save this file? In `ruby/spec` or `mspec` or ?
Can you show what the file looks like in practice?

I was thinking about a list of ignore patterns to be more scalable and require less updates.

Reply:
A bit like we have `backtrace_filter` and `tags_patterns` in MSpec config files currently: https://github.com/ruby/spec/blob/28a0092808856d3e6d5c07f122a082bf2c6cbe46/default.mspec#L36-L45

Reply:
> Can you show what the file looks like in practice?

https://gist.github.com/andrykonchin/c4049ecfa42ae6c79cf255966652df78

> I was thinking about a list of ignore patterns to be more scalable and require less updates.

I guess you say about declaring something like - "the file \<file\> can dynamically create top-level constant \<name\>".

Makes sense. TBH storing just a list of allowed constants seems working well and simpler. But it's up to you.

Reply:
The list is shorter than I thought, nice :)
Did you remove the `include Socket::Constants` in the specs? Otherwise the list would be much longer.
I think we should automatically exclude `/\w+Specs$/` constants as that's the normal pattern for fixtures classes, to shorten the list a bit more.

>  If we compared after and before stages there would be very short list.

But then we would miss constants declared at the top-level. So we need `finish` (+ probably `after` for nice reporting).

> Makes sense. TBH storing just a list of allowed constants seems working well and simpler. But it's up to you.

The list in a file seems nice, let's try that. And then just one exclude pattern `/\w+Specs$/` by default.

> I think patterns/constant names shouldn't be hardcoded in MSpec but in Spec project.

Of course :)